File: error-100-400.js
module.exports = {
  addError(tableau, nombre) {
    if (tableau.includes(nombre)) {
      throw new Error("Erreur 100 : Nombre dupliqué");
    } else if (parseInt(nombre) === 666) {
      throw new Error("Erreur 400 : Nombre sensible");
    } else if (!nombre.match(/^[1-9]\d*$/)) {
      throw new Error("Erreur 400 : Mauvais type de nombre");
    } else if (!nombre.match(/^[1-9]$/)) {
      throw new Error("Erreur 400 : Nombre invalide");
    }
  },
};

File: error_100_400.py
def add_error(tableau, nombre):
    try:
        int_value = int(nombre)
    except ValueError:
        raise ValueError("Erreur 400 : Mauvais type de nombre")

    if nombre in tableau:
        raise ValueError("Erreur 100 : Nombre dupliqué")
    elif int_value == 666:
        raise ValueError("Erreur 400 : Nombre sensible")
    elif int(nombre) == 69:
        raise ValueError("Erreur 451 : Action unavailable due to Legal Reasons")
    elif int(nombre) == 8:
        raise ValueError("Erreur 423 : Lock exception")
    elif not (1 <= int_value <= 10) and (int_value != 0) and (int_value > 10):
        raise ValueError("Erreur 400 : Mauvais type de nombre")
    elif not (1 <= int_value <= 10) and (int_value != 0) and (int_value > 10):
        raise ValueError("Erreur 400 : Nombre invalide")
